docs/claimantmodel/Maps.md

```diff
@@ -0,0 +1,115 @@
+
+Non-canonical forms are not uniquely a problem for maps. Having ambiguous forms could be a problem for Claim Verifiers in the Log Claimant model too, but there Verifiers have the raw Subject and can canonicalize it, or query for substrings/prefixes, etc.
+
+### Map Functions (deriving Claim Subjects from a Claim)
+
+Using a CSM is simplest in the case where there is a single Subject for every Claim. However, in the general case there are multiple Claim Subjects entailed by a Claim. This apparently minor difference has a large impact on the amount of work required in order to maintain discoverability.
+
+```
+mapFn(Claim) []ClaimSubject
+```
+
+We’ll use the signature above when describing how Subjects are derived from Claims. For a Believer to trust that a given Claim is discoverable, they must check that the Claim is committed to under all Subjects returned by mapFn.
+
+The reasoning behind this is best understood through example. If we look at the Subjects entailed by a Certificate for `*.foo.example.co.uk` we might derive:
+ * `foo.example.co.uk.`
+ * `example.co.uk.`
+ * `co.uk.`
+ * `uk.`
+ * `.`
+
+This set of Subjects must have a non-empty intersection with the set of subjects that the Domain Owner is expecting. If this Domain Owner does not have any subdomains and expects to be checking only `example.co.uk.` then a believer convinced only of the inclusion of `foo.example.co.uk.` and no others will have trusted a Claim which cannot be discovered.
+
+This example also highlights that mapFn should not return too many Subjects. While the empty domain `.` may be derived from all Claims, doing so entails a map key which is effectively committing to all entries that would have been in the log. If this is the only mechanism that permits discoverability then a log should have been chosen instead of a map.
+
+### CSM in the Claimant Model
+#### Log-Backed Maps & Mogs
+
+If the map is constructed from a Claim Log then we have the following:
+
+<dl>
+<dt>Claim<sup>CSM</sup></dt>
+<dd><i>"Applying $mapFn to the logged data in $LogCheckpoint results in the map with $rootHash"</i></dd>
+<dt>Statement<sup>CSM</sup></dt>
+<dd>Map Checkpoint (sometimes SMR, or Signed Map Root)</dd>
+<dt>Claimant<sup>CSM</sup></dt>
+<dd>CSM Operator</dd>
+<dt>Believer<sup>CSM</sup></dt>
+<dd>Believer<sup>DOMAIN</sup> and Verifier<sup>DOMAIN</sup></dd>
+<dt>Verifier<sup>CSM</sup></dt>
+<dd>A Verifier<sup>LOG,DOMAIN</sup> with the power to compute the map</dd>
+<dt>Arbiter<sup>CSM</sup></dt>
+<dd>CSM Arbiter, possibly the same as Arbiter<sup>LOG</sup></dd>
+</dl>
+
+#### General Maps & Mogs
+If the map is not constructed from a Claim Log, then the Claim is a lot more nuanced.
+The Map Checkpoint needs to explicitly include a Revision identifier that has a total ordering.
+
+General Map Claim:
```

Comment:
It feels like this is an "Insert-only Map Claim"?
I think we came to the conclusion that in the absence of a log of claimed deltas the General Map claim is probably "I claim this is a map root hash <shrug>", and if it has revisions (either implicitly = there _is_ only one and it's static, or explicitly with rev numbers) then "there's only 1 root hash at revision x"?